### PR TITLE
Feature/fix post delete calls

### PR DESCRIPTION
**Pull request by CaliHam**
What I did:
Implemented post, patch, and delete requests throughout the code. PLEASE NOTE: the API has also been updated and will need to be tested at the same time 

Did this break anything?

- [x] No
- [ ]  Yes

Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x]  Bug fix (non-breaking change which fixes an issue)
- [x]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next?

### CODE REVIEW

File: src/components/TaskListView/TaskListView.js [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+    })
   }
 
-  const rows = savedTasks.sort((a, b) => b.id - a.id).map((savedTask, index) => {
-    return (<TableRow savedTask={savedTask} key={`row-${index}`} deleteTask={deleteTask} savedTasks={savedTasks} setSavedTasks={setSavedTasks} />)
+  const handleChange = (savedTask) => {
+    savedTask.complete = !savedTask.complete
+    patchTask(savedTask.id, savedTask.complete)
+    .then(updatedTasks => setSavedTasks(updatedTasks))
+    .catch(error => console.log(error))
```
Comment (CaliHam) [inline src/components/TaskListView/TaskListView.js:17]: do we want something else to happen in this catch? or is the console.log okay?

**Review by hvohr — APPROVED** [review]
this looks great thank you so much, and should be an easy switch to camel case for mental care

**Review by SethBurns — APPROVED** [review]
Approved pending a solution for the camelCase issue.
Nice work!

File: src/components/TaskListView/TaskListView.js (hunk 1 above)
Comment (taranveersingh93) [reply src/components/TaskListView/TaskListView.js:17]: That's solid. We can add a UI error handling in another ticket

**Review by taranveersingh93 — APPROVED** [review]
Thank you!

**Pull request merged**